sub-packages/bionemo-esm2/src/bionemo/esm2/run/main.py

```diff
@@ -0,0 +1,137 @@
+    def parse_args():
+        parser = argparse.ArgumentParser(description="Run Geneformer pretraining")
+        parser.add_argument("--config", type=str, required=True, help="Path to the JSON configuration file")
+        parser.add_argument(
+            "--model-config-t",
+            default=ExposedESM2PretrainConfig,
+            required=False,
+            help="fully resolvable python import path to the ModelConfig object. Builtin options are ExposedGeneformerPretrainConfig and ExposedFineTuneSeqLenBioBertConfig.",
+        )
+        parser.add_argument(
+            "--data-config-t",
+            default=ESM2DataConfig,
+            required=False,
+            help="fully resolvable python import path to the ModelConfig object.",
+        )
+        parser.add_argument(
+            "--resume-if-exists",
+            default=False,
+            action="store_true",
+            help="Resume training if a checkpoint exists that matches the current experiment configuration.",
+        )
+
+        # Debug options.
+        parser.add_argument(
+            "--nsys-profiling",
+            action="store_true",
+            default=False,
+            help="Enable targeted `nsys` profiling on the training loop for a defined step range. To actually get profiling output you must run the whole program with `nsys`. For example: "
+            " `nsys profile -s none -o output_report_name -t cuda,nvtx --force-overwrite true --capture-range=cudaProfilerApi --capture-range-end=stop  [regular python command here]`",
+        )
+        # start, end, rank
+        parser.add_argument(
+            "--nsys-start-step",
+            type=int,
+            required=False,
+            default=0,
+            help="Start nsys profiling after this step.",
+        )
+        parser.add_argument(
+            "--nsys-end-step",
+            type=int,
+            required=False,
+            help="End nsys profiling after this step.",
+        )
+        # rank as list of integers
+        parser.add_argument(
+            "--nsys-ranks",
+            type=int,
+            nargs="+",
+            required=False,
+            default=[0],
+            help="Enable nsys profiling for these ranks.",
+        )
+        return parser.parse_args()
+
+    def string_to_class(path: str):
+        import importlib
+
+        module_path, class_name = path.rsplit(".", 1)
+        module = importlib.import_module(module_path)
+        return getattr(module, class_name)
+
+    def load_config(config_path: str, model_config_t: Optional[str], data_config_t: Optional[str]) -> MainConfig:
+        with open(config_path, "r") as f:
+            config_dict = json.load(f)
+
+        # model/data_config_t is used to select the parser dynamically.
+        if model_config_t is None or model_config_t == "ExposedESM2PretrainConfig":
+            model_config_t = ExposedESM2PretrainConfig
+        elif model_config_t == "ExposedFineTuneSeqLenBioBertConfig":
```

Comment:
The finetune config class names are different in ESM2. Can we support those instead of Geneformer finetune config class?

Reply:
https://github.com/NVIDIA/bionemo-framework/blob/48c02c3555ccc8d65d1d10ffed9f4c1f7ec50b63/sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/finetune_regressor.py#L118

and 

https://github.com/NVIDIA/bionemo-framework/blob/48c02c3555ccc8d65d1d10ffed9f4c1f7ec50b63/sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/finetune_token_classifier.py#L140

Reply:
Oh I dont think I actually exposed these...

Reply:
Let's get these in the next release.